File: simplify-path/simplify-path.js
var simplifyPath = function(path) {
    const pathSplit = path.split('/');
    const result = [];
    for (let i = 0; i < pathSplit.length; i++) {
        if (pathSplit[i] === '..' && result.length > 0) {
            result.pop()
        } else if (pathSplit[i] !== '' && pathSplit[i] !== '.' && pathSplit[i] !== '..') {
            result.push(pathSplit[i])
        }
    };
    console.log(result);
    return `/${result.join('/')}`
};

File: simplify-path/simplify-path.py
#https://leetcode.com/problems/simplify-path/

class Solution:
    def simplifyPath(self, path: str) -> str:
        path_split = path.split('/')
        result = []
        for s in path_split:
            if s == '..' and len(result) > 0:
                result.pop()
            elif s != '.' and s != '..' and s != '':
                result.append(s)
        return f'/{"/".join(result)}'